packages/next-intl/test/react-server/index.test.tsx

```diff
@@ -57,4 +71,22 @@ describe('performance', () => {
     // Render 3: Recursive call that renders synchronously as well
     expect(renderCount).toBe(3);
   });
+
+  it('shares message format cache between useTranslations and getTranslations', async () => {
+    // First invocation
+    useTranslations('Component');
+    const firstCallCache =
+      vi.mocked(createTranslator).mock.calls[0][0].messageFormatCache;
+
+    // Second invocation with a different namespace
+    await getTranslations('Component2');
+    const secondCallCache =
+      vi.mocked(createTranslator).mock.calls[1][0].messageFormatCache;
+
+    // Verify that the same cache instance is used in both invocations
+    expect(firstCallCache).toBe(secondCallCache);
```

Comment:
It seems like the test accidentally passes currently despite us having two separate caches:

1. https://github.com/fkapsahili/next-intl/blob/f1fc3ddd9f1b2919ca0b750c0eda682477bcd4f7/packages/next-intl/src/server/react-server/getTranslations.tsx#L17

2. https://github.com/fkapsahili/next-intl/blob/f1fc3ddd9f1b2919ca0b750c0eda682477bcd4f7/packages/next-intl/src/react-server/getTranslator.tsx#L15

That is due to way the mock implementation of `cache` currently works:

https://github.com/amannn/next-intl/blob/716d71605c2b855c6c6004e0c233bbc20ddeaf74/packages/next-intl/__mocks__/react.tsx#L29-L31

Maybe we should use a `WeakMap` for the cache key there to make sure we use a different cache for different functions?

Based on this, this test should hopefully fail then and we can fix it by actually using the same message format cache instance.

Does that make sense?

Reply:
Thanks for pointing this out - this definitely makes sense! The current mock setup indeed leads to the situation where the test passes despite the usage of separate cache implementations.

Using a WeakMap for the cache key, worked perfectly as you suggested. I think this will help ensure that the cache behaves more realistically in the test environment. This change also made the test fail as expected 🙂.

I've now added an implementation using the `WeakMap` in the mock cache and added a shared cache instance using `messageFormatCache` to make sure both `getTranslations` and `useTranslations` actually use the same message format cache instance, as intended. I've updated the PR with these improvements. I thought it would make sense to locate the shared cache in the new file `messageFormatCache.tsx` in the `shared` directory.

Let me know if there is anything to improve! 🙂